roles/upload_iso_to_datastore/README.md

```diff
@@ -0,0 +1,115 @@
+    - The vSphere vCenter or ESXi host password.
+    - If this variable is not set, the collection level variable `vmware_ops_password` will be used. If that variable is not set, the environment variable `VMWARE_PASSWORD` will be used. At least one of these variables must be set to use this role.
+    - See the [authentication documentation](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/docs/authentication.md) for examples.
+
+- **upload_iso_to_datastore_validate_certs** (bool)
+    - Allows connection when SSL certificates are not valid. Set to false when certificates are not trusted.
+    - If this variable is not set, the collection level variable `vmware_ops_validate_certs` will be used. If that variable is not set, the environment variable `VMWARE_VALIDATE_CERTS` will be used.
+    - See the [authentication documentation](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/docs/authentication.md) for examples.
+
+- **upload_iso_to_datastore_port** (int or str)
+    - The port used to authenticate to the vSphere vCenter or ESXi host.
+    - If this variable is not set, the collection level variable `vmware_ops_port` will be used. If that variable is not set, the environment variable `VMWARE_PORT` will be used.
+    - See the [authentication documentation](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/docs/authentication.md) for examples.
+
+### Proxy
+
+- **upload_iso_to_datastore_proxy_host** (str)
+    - The hostname of a proxy host that should be used for all HTTPs communication by the role.
+    - The format is a hostname or an IP.
+    - If this variable is not set, the collection level variable `vmware_ops_proxy_host` will be used. If that variable is not set, the environment variable `VMWARE_PROXY_HOST` will be used.
+    - See the [authentication documentation](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/docs/authentication.md) for examples.
+
+- **upload_iso_to_datastore_proxy_port** (str or int)
+    - The port of a proxy host that should be used for all HTTPs communication by the role
+    - If this variable is not set, the collection level variable `vmware_ops_proxy_host` will be used. If that variable is not set, the environment variable `VMWARE_PROXY_PORT` will be used.
+    - See the [authentication documentation](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/docs/authentication.md) for examples.
+
+### Uploading an ISO file
+
+- **upload_iso_to_datastore_src** (str, required)
+    - Absolute path to the ISO file to upload.
+    - This parameter is case sensitive.
+
+- **upload_iso_to_datastore_datacenter** (str, required)
+    - The name of the datacenter where the Datastore resides.
+    - This parameter is case sensitive.
+
+- **upload_iso_to_datastore_datastore** (str, required)
+    - The ISO will be uploaded into this Datastore.
+    - Please see the examples for more usage.
+
+- **upload_iso_to_datastore_dst_folder** (str, required)
```

Comment:
Is this parameter mandatory? Or can it can be set to an empty string to create the ISO in the root?

Reply:
Previously I only thought about adding a `folder` var to a test, not to a role itself.